src/pudl/analysis/pudl_models.py

```diff
@@ -48,7 +51,179 @@ def core_sec10k__quarterly_company_information() -> pd.DataFrame:
 
 @asset(
     io_manager_key="pudl_io_manager",
-    group_name="pudl_models",
+    group_name="sec10k",
+)
+def core_sec10k__quarterly_company_information(
+    raw_sec10k__quarterly_company_information: pd.DataFrame,
+) -> pd.DataFrame:
+    """Company information extracted from SEC10k filings.
+
+    Consolidate company information extracted from key: value blocks
+    within the headers of the SEC 10k filings such that there
+    is one record of company information per block. One company's
+    information may be reported in multiple filings on the same report date.
+    However, we only want to keep one of those records per company and report date,
+    (``central index key`` and ``report_date`` are a primary key for the table).
+    We prioritize keeping records from filings where that company's extracted
+    ``central_index_key`` matches the filer's central index key, meaning that
+    that company filed the 10k itself.
+    """
+    # Strip erroneous "]" characters
+    raw_sec10k__quarterly_company_information["company_information_fact_name"] = (
+        raw_sec10k__quarterly_company_information[
+            "company_information_fact_name"
+        ].str.lstrip("]")
+    )
+    df = raw_sec10k__quarterly_company_information.pivot(
+        values="company_information_fact_value",
+        index=[
+            "filename_sec10k",
+            "report_date",
+            "company_information_block",
+            "company_information_block_count",
+        ],
+        columns="company_information_fact_name",
```

Comment:
I added `company_information_block` and `company_information_block_count` to the index, then I groupby filename, report date, and company information block count to consolidate the information into one record per block.